foyer/general_forcefield.py

```diff
@@ -0,0 +1,564 @@
+                self.atomTypeClasses[name] = atype.atomclass
+                if atype.tags.get("element"):
+                    ele = atype.tags["element"]
+                    if element.element_by_symbol(ele):
+                        self.atomTypeElements[atype.name] = ele
+                    else:
+                        self.non_element_types[ele] = None
+                else:
+                    # Register atomtype with missing atomtype as atomistic (empty string)
+                    self.atomTypeElements[name] = ""
+        return None
+
+    def apply(
+        self,
+        top,
+        references_file=None,
+        use_residue_map=True,
+        assert_bond_params=True,
+        assert_angle_params=True,
+        assert_dihedral_params=True,
+        assert_improper_params=True,
+        combining_rule="geometric",
+        debug=False,
+        *args,
+        **kwargs,
+    ):
+        """Apply the force field to a molecular topology.
+
+        Parameters
+        ----------
+        top : gmso.Topology, or mb.Compound
+            Molecular Topology to apply the force field to
+        references_file : str, optional, defaut=None
+            Name of file where force field references will be written
+            (in Bibtex format).
+        use_residue_map : bool, optional, default=True
+            Options to speed up if there are a lot of repeated
+            subtopology within the topology (assuming they all
+            have the same name).
+        assert_bond_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system bonds.
+        assert_angle_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system angles.
+        assert_dihedral_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system dihedrals.
+        assert_improper_params : bool, optional, default=True
+            If True, Foyer will exit if parameters are not found for all
+            system impropers.
+        combining_rule : str, optional, default='geometric'
+            The combining rule of the system, stored as an attribute of the
+            GMSO topology. Available options: 'geometric', 'lorentz'.
+        debug : bool, optional, default=False
+            If True, Foyer will print debug-level information about notable or
+            potential problematic details it encounters.
+        """
+        if self.atomTypeDefinitions == {}:
+            raise FoyerError(
+                "Attempting to atom-type using a forcefield "
+                "with no atom type definitions."
+            )
+
+        if self.backend == "gmso":
+            if not isinstance(top, gmso.Topology):
+                mb = import_("mbuild")
+                tmp_top = mb.load(top)
+                top = gmso.external.from_mbuild(tmp_top)
```

Comment:
What kind of object/files are supported by `mbuild.load`? It might be nice to do the check?